votor-messages/src/vote.rs

```diff
@@ -0,0 +1,263 @@
+//! Vote data types for use by clients
+use {
+    serde::{Deserialize, Serialize},
+    solana_clock::Slot,
+    solana_hash::Hash,
+};
+
+/// Enum that clients can use to parse and create the vote
+/// structures expected by the program
+#[cfg_attr(
+    feature = "frozen-abi",
+    derive(AbiExample, AbiEnumVisitor),
+    frozen_abi(digest = "FRn4f3PTtbvw3uv2r3qF8K49a5UF4QqDuVdyeshtipTW")
+)]
+#[derive(Clone, Copy, Debug, PartialEq, Serialize, Deserialize)]
+pub enum Vote {
+    /// A notarization vote
+    Notarize(NotarizationVote),
```

Comment:
Can we pleeeease not step on this landmine again and use TLV instead of a enum here? This will inevitably turn into maintenance nightmare the moment we will want to change format of any of these messages, same as in gossip.